lib/internal/crypto/random.js

```diff
@@ -291,20 +294,43 @@ function getRandomValues(data) {
 // Implements an RFC 4122 version 4 random UUID.
 // To improve performance, random data is generated in batches
 // large enough to cover kBatchSize UUID's at a time. The uuidData
-// and uuid buffers are reused. Each call to randomUUID() consumes
-// 16 bytes from the buffer.
-
-const kHexDigits = [
-  48, 49, 50, 51, 52, 53, 54, 55,
-  56, 57, 97, 98, 99, 100, 101, 102
-];
+// buffer is reused. Each call to randomUUID() consumes 16 bytes
+// from the buffer.
 
 const kBatchSize = 128;
 let uuidData;
 let uuidNotBuffered;
-let uuid;
 let uuidBatch = 0;
 
+let hexBytesCache;
+function getHexBytes() {
+  if (hexBytesCache === undefined) {
+    hexBytesCache = new Array(256);
+    for (let i = 0; i < hexBytesCache.length; i++) {
+      const hex = NumberPrototypeToString(i, 16);
+      hexBytesCache[i] = StringPrototypePadStart(hex, 2, '0');
+    }
+  }
+  return hexBytesCache;
+}
+
+function serializeUUID(buf, offset = 0) {
+  const kHexBytes = getHexBytes();
```

Comment:
To simplifify things a bit further here ... just generate the hex array on module load...

```suggestion
let uuidBatch = 0;

const kHexBytes = new Array(256);
for (let i = 0; i < kHexBytes.length; i++) {
  const hex = NumberPrototypeToString(i, 16);
  kHexBytes[i] = StringPrototypePadStart(hex, 2, '0');
}

function serializeUUID(buf, offset = 0) {
```

Reply:
Thanks! I like this simplification (4th variant), but it reserves 2KB of data immediately after loading `crypto`. I couldn't imagine a case where it could be a problem, but - for safety - I introduced a lazy getter to avoid that.

Just to confirm, does it mean that this 2KB allocation is negligible? :)

Reply:
On the other hand - when the `kHexBytes` is initialized statically without the `for` loop, the results are ~8% better.

Do you think that it's worth speeding it up this way instead?

```js
const kHexBytes = [
  '00', '01', '02', '03', '04', '05', '06', '07', '08', '09', '0a', '0b', '0c', '0d', '0e', '0f',
  '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '1a', '1b', '1c', '1d', '1e', '1f',
  '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '2a', '2b', '2c', '2d', '2e', '2f',
  '30', '31', '32', '33', '34', '35', '36', '37', '38', '39', '3a', '3b', '3c', '3d', '3e', '3f',
  '40', '41', '42', '43', '44', '45', '46', '47', '48', '49', '4a', '4b', '4c', '4d', '4e', '4f',
  '50', '51', '52', '53', '54', '55', '56', '57', '58', '59', '5a', '5b', '5c', '5d', '5e', '5f',
  '60', '61', '62', '63', '64', '65', '66', '67', '68', '69', '6a', '6b', '6c', '6d', '6e', '6f',
  '70', '71', '72', '73', '74', '75', '76', '77', '78', '79', '7a', '7b', '7c', '7d', '7e', '7f',
  '80', '81', '82', '83', '84', '85', '86', '87', '88', '89', '8a', '8b', '8c', '8d', '8e', '8f',
  '90', '91', '92', '93', '94', '95', '96', '97', '98', '99', '9a', '9b', '9c', '9d', '9e', '9f',
  'a0', 'a1', 'a2', 'a3', 'a4', 'a5', 'a6', 'a7', 'a8', 'a9', 'aa', 'ab', 'ac', 'ad', 'ae', 'af',
  'b0', 'b1', 'b2', 'b3', 'b4', 'b5', 'b6', 'b7', 'b8', 'b9', 'ba', 'bb', 'bc', 'bd', 'be', 'bf',
  'c0', 'c1', 'c2', 'c3', 'c4', 'c5', 'c6', 'c7', 'c8', 'c9', 'ca', 'cb', 'cc', 'cd', 'ce', 'cf',
  'd0', 'd1', 'd2', 'd3', 'd4', 'd5', 'd6', 'd7', 'd8', 'd9', 'da', 'db', 'dc', 'dd', 'de', 'df',
  'e0', 'e1', 'e2', 'e3', 'e4', 'e5', 'e6', 'e7', 'e8', 'e9', 'ea', 'eb', 'ec', 'ed', 'ee', 'ef',
  'f0', 'f1', 'f2', 'f3', 'f4', 'f5', 'f6', 'f7', 'f8', 'f9', 'fa', 'fb', 'fc', 'fd', 'fe', 'ff'
];
```